ui5/eve7/controller/EveTable.controller.js

```diff
@@ -197,61 +198,118 @@ sap.ui.define([
       },
 
       toggleTableEdit: function() {
-
-         var header = this.getView().byId("header");
+         var header = sap.ui.getCore().byId("EveViewer21--header");
          if (!this.editor) {
             this.editor = new VerticalLayout("tableEdit", {"width":"100%"});
 
             header.addContent(this.editor);
 
             // expression row
             {
-               var collection = this.mgr.GetElement(this.eveTable.fCollectionId);
-               var oModel = new JSONModel();
-               oModel.setData(collection.publicFunction);
-               // oModel.setData(aData);
-               this.getView().setModel(oModel);
-
-               var exprIn = new mInput("expression", {
-                                         width:"98%",
-                                         type : sap.m.InputType.Text,
-                                         placeholder: "Expression",
-                                         showSuggestion: true
-                                       });
+               var collection = this.mgr.GetElement(this.eveTable.fCollectionId);               
+               var exprIn = new sap.m.Input("inputExp", {
+    		  placeholder: "Start expression with \"i.\" to access object",
+    		  showValueHelp: true,
+    		  showTableSuggestionValueHelp: false,
+    		  width: "100%",
+    		  maxSuggestionWidth: "500px",
+    		  showSuggestion: true,
+    		  valueHelpRequest: function (oEvent) {
+    		     sap.m.MessageBox.alert("Write any valid expression.\n Using \"i.\" convetion to access an object in collection. Below is an example:\ni.GetPdgCode() + 2");
```

Comment:
Use MessageBox instead of sap.m.MessageBox

Reply:
I have made the correction. Thanks!